dbos/_dbos_config.py

```diff
@@ -143,21 +312,26 @@ def load_config(
 
     init_logger()
 
-    with open(config_file_path, "r") as file:
-        content = file.read()
-        substituted_content = _substitute_env_vars(content)
-        data = yaml.safe_load(substituted_content)
+    data = parse_config_file(config_file_path)
 
-    # Load the JSON schema relative to the package root
-    schema_file = resources.files("dbos").joinpath("dbos-config.schema.json")
-    with schema_file.open("r") as f:
-        schema = json.load(f)
+    config: ConfigFile = process_config(data=data, silent=silent)
+    # Check the connectivity to the database and make sure it's properly configured
+    # Note, never use db wizard if the DBOS is running in debug mode (i.e. DBOS_DEBUG_WORKFLOW_ID env var is set)
+    debugWorkflowId = os.getenv("DBOS_DEBUG_WORKFLOW_ID")
+    if use_db_wizard and debugWorkflowId is None:
+        config = db_wizard(config, config_file_path)
 
-    # Validate the data against the schema
-    try:
-        validate(instance=data, schema=schema)
-    except ValidationError as e:
-        raise DBOSInitializationError(f"Validation error: {e}")
+    if "local_suffix" in config["database"] and config["database"]["local_suffix"]:
+        config["database"]["app_db_name"] = f"{config['database']['app_db_name']}_local"
```

Comment:
I like these outside of `process_config`. When passing an object explicitly, I am expecting library users to already have configured their database -- they give in their DB parameters, after all. This method is only called when no config is passed and we load the config from `dbos-config.yaml`